### PR TITLE
Dump report after each protocol

### LINKED ISSUE #436: Generate report after each routine

### PR DESCRIPTION
**Pull request by andrea-pasquale**
As title says, closes #436

Checklist:
- [ ] Reviewers confirm new code works as expected.
- [ ] Tests are passing.
- [ ] Coverage does not decrease.
- [ ] Documentation is updated.

### CODE REVIEW

**Review by Jacfomg — APPROVED** [review]
Look fine to me. I also tested it in hardware.

File: src/qibocal/protocols/characterization/resonator_spectroscopy.py [hunk 1]
```diff
@@ -61,6 +68,22 @@ class ResonatorSpectroscopyResults(Results):
...
+    @property
+    def update(self):
+        """Method overwritten from Results to not update
+        amplitude when running resonator spectroscopy at
+        high power."""
+        up: dict[str, ParameterValue] = {}
+        if self.bare_frequency == {}:
+            fields_to_updated = [fld for fld in fields(self) if fld.name != "amplitude"]
+        else:
+            fields_to_updated = fields(self)
```
Comment (rodolfocarobene) [inline src/qibocal/protocols/characterization/resonator_spectroscopy.py:80]:
```suggestion
        fields_to_updated = [fld for fld in fields(self) if fld.name != "amplitude"] if self.bare_frequency == {} else fields(self)

```

Comment (rodolfocarobene) [reply src/qibocal/protocols/characterization/resonator_spectroscopy.py:80]: might be too long... not sure

**Review by rodolfocarobene — APPROVED** [review]

File: src/qibocal/protocols/characterization/resonator_spectroscopy.py (hunk 1 above)
Comment (andrea-pasquale) [reply src/qibocal/protocols/characterization/resonator_spectroscopy.py]: I like it

**Pull request merged**